docs/file-formats/kmx-file-format.md

```diff
@@ -232,6 +234,58 @@ compiler expands `any()` statements found in the key part of a rule into
 multiple rules -- so the key part of the rule is always only ever matching a
 single key combination.
 
+`ShiftFlags` contains a set of modifier and state key flags, and some additional
+flags that determine how to interpret `Key`.
+
+### Modifier and state key flags
+
+The modifier and state key flags are:
```

Comment:
For this section and what follows, we probably could just reference `common/web/types/src/consts/modifier-key-constants.ts`, couldn't we?

There _is_ a bit of extra clarification on a few of the trickier points, admittedly, but it's low-quantity enough that we could just inline those clarifications within that source file.

Alternatively, we could just toss a link to the file _and_ keep the section otherwise as-is.

Reply:
> For this section and what follows, we probably could just reference `common/web/types/src/consts/modifier-key-constants.ts`, couldn't we?

No. The entire purpose of a specification is to be a stable single-point reference document, no matter what happens to the code. The code should be implemented on the basis of the specification, not the other way around (and I know that this spec has been written post-facto, but that's not the ideal).

I am thinking I should add all the virtual key codes to this document as currently it references the online help for those, which does not have the same stability guarantee that we will give this spec.

(If I was going to point to a source file, I'd choose the origin of the values (kmx_file.h), not the .ts translation of it.)